modules/dop/providers/projectpipeline/service_impl.go

```diff
@@ -391,6 +392,13 @@ func (p *ProjectPipelineService) List(ctx context.Context, params deftype.Projec
 		Status:      params.Status,
 		AscCols:     params.AscCols,
 		DescCols:    params.DescCols,
+		IsOthers:    params.CategoryKey == apistructs.CategoryOthers,
+		FilePathWithNames: func() []string {
+			if params.CategoryKey != apistructs.CategoryOthers {
+				return apistructs.CategoryKeyRuleMap[apistructs.PipelineCategory(params.CategoryKey)]
+			}
+			return append(apistructs.CategoryKeyRuleMap[apistructs.CategoryBuildDeploy], apistructs.CategoryKeyRuleMap[apistructs.CategoryBuildArtifact]...)
+		}(),
```

Comment:
separate a func like: `getRulesByCategoryKey()`, and add unit test.

Reply:
done